samples/cpp/hello_reshape_ssd/main.cpp

```diff
@@ -2,214 +2,149 @@
-        // ----------------------------------------------------
-        DataPtr output_info;
-        std::string output_name;
-        std::tie(output_name, output_info) = *outputs_info.begin();
-        // SSD has an additional post-processing DetectionOutput layer
-        // that simplifies output filtering, try to find it.
-        if (auto ngraphFunction = network.getFunction()) {
-            for (const auto& op : ngraphFunction->get_ops()) {
-                if (op->get_type_info() == ngraph::op::DetectionOutput::get_type_info_static()) {
-                    if (output_info->getName() != op->get_friendly_name()) {
-                        throw std::logic_error("Detection output op does not produce a network output");
-                    }
-                    break;
-                }
-            }
-        }
 
-        const SizeVector output_shape = output_info->getTensorDesc().getDims();
-        const size_t max_proposal_count = output_shape[2];
-        const size_t object_size = output_shape[3];
-        if (object_size != 7) {
-            throw std::logic_error("Output item should have 7 as a last dimension");
-        }
-        if (output_shape.size() != 4) {
-            throw std::logic_error("Incorrect output dimensions for SSD model");
-        }
-        if (output_info == nullptr) {
-            IE_THROW() << "[SAMPLES] internal error - output information is empty";
-        }
+        std::shared_ptr<unsigned char> image_data = reader->getData();
+        size_t image_channels = 3;
+        size_t image_width = reader->width();
+        size_t image_height = reader->height();
 
-        output_info->setPrecision(Precision::FP32);
-
-        auto dumpVec = [](const SizeVector& vec) -> std::string {
-            if (vec.empty())
-                return "[]";
-            std::stringstream oss;
-            oss << "[" << vec[0];
-            for (size_t i = 1; i < vec.size(); i++)
-                oss << "," << vec[i];
-            oss << "]";
-            return oss.str();
-        };
-        std::cout << "Resulting input shape = " << dumpVec(input_shape) << std::endl;
-        std::cout << "Resulting output shape = " << dumpVec(output_shape) << std::endl;
-        // -----------------------------------------------------------------------------------------------------
+        // reshape model to image size and batch size
+        ov::Shape tensor_shape = model->input().get_shape();
 
-        // --------------------------- Step 4. Loading a model to the device
-        // ------------------------------------------
-        ExecutableNetwork executable_network = ie.LoadNetwork(network, device_name);
-        // -----------------------------------------------------------------------------------------------------
+        tensor_shape[0] = batch_size;
+        tensor_shape[1] = image_channels;
+        tensor_shape[2] = image_height;
+        tensor_shape[3] = image_width;
```

Comment:
suggest to use `tensor_layout` here. E.g. `ov::layout::batch_idx(tensor_layout)` instead of hardcoded `0`